CONTRIBUTING.md

```diff
@@ -56,83 +56,185 @@ To install Agda locally and use that install with emacs, you can do the followin
+For instructions to setup Agda without nix, check [Setup without
+nix](#setup-without-nix).
 
-For instructions _not using_ `nix-shell`, check [Setup without nix-shell](setup-without-nix-shell).
+To typecheck the formal specification, run:
 
-## Working on libraries
+```
+agda src/Everything.agda
+```
 
-To work simultaneously on the ledger and one of its dependencies, the easiest way to do this is to remove the library from the ledger's `.agda-lib` file and add its path to the `include:` section. Then, when finished, push the changes to the library, then update `default.nix` to point to your new commit.
+### Modifying the Agda libraries
 
-Warning: Don't forget to update the SHA when updating `default.nix`. Nix will fail silently on your local machine if you do that. Just change a few characters, run `nix-build -A ledger` and nix will tell you the correct hash to put there.
+To work simultaneously on the ledger and one of its dependencies, the easiest
+way to do this is to remove the library from the ledger's `.agda-lib` file and
+add its path to the `include:` section. 
 
-## Building the PDF quickly
+When finished, push the changes to the library and update `default.nix` to point
+to your new commit.
 
-The Makefile can be used to build the PDF without having to build everything else. Either run `make` from within `nix-shell`, or use
-```
-nix-shell --command 'make docs'
-```
-to run `make` without launching an interactive shell.
+*Warning:* Don't forget to update the SHA when updating `default.nix`. Nix will
+fail silently on your local machine if you do that. Just change a few
+characters, run `nix-build -A ledger` and nix will tell you the correct hash to
+put there.
 
-This combines well with the ability to invoke the TeX backend of Agda within Emacs,
-which is much faster when you have already loaded an Agda file/interface.
+## Building the artifacts
 
-## Building other artifacts
+`formal-ledger-specifications` is built using a custom system written in the
+Haskell DSL [Shake](https://shakebuild.com/). The file
+[`Shakefile.hs`](Shakefile.hs) contains the source code of the build system. We
+refer to the build-system binary as `fls-shake`.
 
-Apart from the PDF specification, the `Makefile` can be used to also generate the following:
-- `make html`: generate HTML under `dist/html/`
-- `make hs`: generate Haskell code under `dist/MAlonzo/`
-- `make hsTest`: run the Haskell tests of each Agda formalisation
-- `make staticWebsite`: gather all resources above in a central webpage `dist/index.html`
+Depending on whether one uses nix or not, the commands given in the rest of this
+instructions are to be run differently:
 
-## Updating nixpkgs
+- For nix users, the commands should be executed using nix shell via:
+
+    ```
+    nix-shell --run COMMAND
+    ```
+
+    This ensures that the correct dependencies are in scope.
+
```

Comment:
It's not entirely clear what you mean here.  Do you mean "follow the instructions below, but where you see `fls-shake COMMAND`, replace `fls-shake` with `nix-shell --run`?  I'm pretty sure that's not what you mean, since `nix-shell --run conway-ledger.pdf` surely doesn't work... nor does `nix-shell --run "make conway-ledger.pdf"`, for example.  Maybe give at least one example of what you mean.  That should make it clear enough.

Reply:
I added an example